meteor/sofie-server/data-access/repositories/mongo/mongo-rundown-repository.ts

```diff
@@ -49,4 +53,33 @@ export class MongoRundownRepository extends BaseMongoRepository implements Rundo
 	public saveRundown(_rundown: Rundown): void {
 		throw new Error('Not implemented')
 	}
+
+	public async deleteRundown(rundownId: string): Promise<void> {
+		this.assertDatabaseConnection(this.deleteRundown.name)
+		await this.assertRundownExist(rundownId)
+		await this.segmentRepository.deleteSegmentsForRundown(rundownId)
+
+		const rundownDeletionResult: DeleteResult = await this.getCollection().deleteOne({
+			_id: rundownId,
+		})
+		if (!rundownDeletionResult.acknowledged) {
+			throw new DeletionFailedException(`Deletion of rundown was not acknowledged, for rundownId: ${rundownId}`)
+		}
+		if (rundownDeletionResult.deletedCount === 0) {
+			throw new DeletionFailedException(
+				`Expected to delete one rundown, but none was deleted, for rundownId: ${rundownId}`
+			)
+		}
+	}
+	private async assertRundownExist(rundownId: string): Promise<void> {
+		const findCursor = this.getCollection().find({
+			_id: rundownId,
+		})
+		const mongoRundown: MongoRundown = (
+			await findCursor.project({ _id: 1 }).toArray()
+		)[0] as unknown as MongoRundown
```

Comment:
You don't gain much by separating this into local variables.
It's completely fine to chain them:
```
this.getCollection()
  .find({ _id: rundownId })
  .project({ _id: 1 })
  .toArray()
```
But `Collection` has a `countDocuments` method that returns a `number`, so we could just write:
```
const doesRundownExist: boolean = (await this.getCollection.countDocuments({ _id: rundownId })) === 1
```

Only just discovered this method, which is why I didn't suggest it before

Reply:
I have used the `countDocuments` during tests, but didn't notice that it could take a filter.
Have updated it to use that way of checking if the rundown in question exists.

Reply:
Did you push the changes?

Reply:
Now i have :)